### PR TITLE
Add Identityproof tests

### PR DESCRIPTION
**Pull request by rullzer**
* Add tests for Key
* Add tests for Manager
* Add tests for Signer
* Removed URLGenerator from signer

@LukasReschke

### CODE REVIEW

File: lib/private/Security/IdentityProof/Signer.php [hunk 1]
```diff
@@ -100,17 +81,17 @@ public function verify(array $data) {
 			&& isset($data['signature'])
 			&& isset($data['message']['signer'])
 		) {
-			$server = $this->urlGenerator->getAbsoluteURL('/');
-			$postfix = strlen('@' . rtrim($this->removeProtocolFromUrl($server), '/'));
-			$userId = substr($data['message']['signer'], -$postfix);
+			$location = strrpos($data['message']['signer'], '@');
```
Comment (LukasReschke) [inline lib/private/Security/IdentityProof/Signer.php]: Doesn't cover cases like `lukas@nextcloud.com@cloud.nextcloud.com` 🙈

**Review by LukasReschke — CHANGES_REQUESTED** [review]

File: lib/private/Security/IdentityProof/Signer.php (hunk 1 above)
Comment (rullzer) [inline lib/private/Security/IdentityProof/Signer.php]: Yes it does. becase then your usename is `lukas@nextcloud.com` and not `lukas`.

Comment (LukasReschke) [inline lib/private/Security/IdentityProof/Signer.php]: Oh. True 🙈 – My bad.

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
LGTM

**Comment by rullzer** [comment]
CC: @MorrisJobke @nickvergessen @icewind1991

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**